packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -386,24 +389,37 @@ class ScheduleManagerCore {
             assert(sequenceNumber < this.pauseSequenceNumber, "we should never start processing incomplete batch!");
             // If the next op is the start of incomplete batch, then we can't process it until it's fully in - pause!
             if (sequenceNumber + 1 === this.pauseSequenceNumber) {
-                this.setPaused(true);
+                this.setPaused();
             }
         }
     }
 
-    private setPaused(localPaused: boolean) {
+    private setPaused() {
+        assert(!this.localPaused, "always called from resumed state");
+        this.localPaused = true;
+        this.timePaused = performance.now();
+        // eslint-disable-next-line @typescript-eslint/no-floating-promises
+        this.deltaManager.inbound.pause();
```

Comment:
Is it okay to not await this? It seems risky that pause() is async. What if the queue is paused later and some ops pass through?

Reply:
That should not be possible in UT, as all of the test cases assume that UT is the only entity controlling pausing, and we know we use mock implementation that pauses it right away. But I'll add awaits - it does not change anything